figure_generation/02_results_table_generation.py

```diff
@@ -0,0 +1,462 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:light
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:snorkeling_full_text]
+#     language: python
+#     name: conda-env-snorkeling_full_text-py
+# ---
+
+# # Top Ten Predicted Sentences for each Edge Type
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+# %matplotlib inline
+
+from pathlib import Path
+
+import pandas as pd
+import plydata as ply
+from snorkel.labeling.model import LabelModel
+from sqlalchemy import create_engine
+
+# +
+# Set up the environment
+username = "danich1"
+password = "snorkel"
+dbname = "pubmed_central_db"
+
+# Path subject to change for different os
+database_str = (
+    f"postgresql+psycopg2://{username}:{password}@/{dbname}?host=/var/run/postgresql"
+)
+conn = create_engine(database_str)
+# -
+
+gen_performance_list = list(
+    Path("../generative_model_training/output/performance").rglob("*performance.tsv")
+)
+gen_training_data_list = list(
+    Path("../generative_model_training/output").rglob("*temp_abstract_file.tsv")
+)
+training_candidate_label_matrix = list(
+    Path("../label_candidates/output").rglob(
+        "*abstract_train_candidates_resampling.tsv"
+    )
+)
+predicted_sentences_list = list(
+    Path("../edge_prediction_experiment/output").rglob("*sentences.tsv")
+)
+
+gen_model_best_params = dict()
+for dataframe_file in gen_performance_list:
+    edge_label = dataframe_file.stem.split("_")[0]
+    performance_df = pd.read_csv(dataframe_file, sep="\t")
+    best_params = (
+        performance_df
+        >> ply.query("data_source=='abstract'")
+        >> ply.query(f"label_source=='{edge_label}'")
+        >> ply.query("lf_num==lf_num.max()")
+        >> ply.query("model=='tune'")
+        >> ply.select("epochs", "l2_param", "lr_param")
+        >> ply.distinct()
+    ).values[0]
+    gen_model_best_params[edge_label.lower()] = list(best_params)
+
+candidate_label_matrices = dict()
+for label_matrix_datafile in training_candidate_label_matrix:
+    edge_label = label_matrix_datafile.stem.split("_")[0]
+    edge_label = (
+        "dag"
```

Comment:
I don't think I've ever seen nested inline if/else statements like this. Does it do what you'd expect it to do?

Reply:
Got it! I've seen ternary operators before but I've never seen them chained like this (although it makes sense that they would work). Good to know!